tests/unit/adapters/application-test.js

```diff
@@ -29,103 +29,187 @@ moduleFor('adapter:application', 'Unit | Adapter | application', {
 });
 
 test('#find calls #findOne when options arg is a string', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
   let promise = adapter.find('1');
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(adapter.findOne.calledWith('1'), 'findOne called with "1"');
+    done();
+  });
 });
 
 test('#find calls #findOne when options arg is an object having an id property', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject();
   sandbox.stub(adapter, 'findOne', function () { return RSVP.Promise.resolve(null); });
-  let options = { id: '1', query: {sort: '-date'} };
+  let options = {id: '1', query: {sort: '-date'}};
   let promise = adapter.find(options);
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(adapter.findOne.calledOnce, 'findOne called');
-  assert.ok(adapter.findOne.calledWith(options.id, options.query), 'findOne called with `"1"` and query `{"sort": "-date"}`');
+  promise.then(() => {
+    assert.ok(adapter.findOne.calledOnce, 'findOne called');
+    assert.ok(
+      adapter.findOne.calledWith(options.id, options.query),
+      'findOne called with `"1"` and query `{"sort": "-date"}`'
+    );
+    done();
+  });
 });
 
+// Yeah... findQuery is used when options arg is not a string or an object.
+// undefined is not a string or an object, but so is a number.
+// TODO: id.toString() in to public methods? Or make this test more specific?
```

Comment:
In theory I am all for sticking to the specs. Less code. Less bugs.

This case however will end up annoying users a lot. `store.find('post', 1)` can only mean one thing, get `post` with id `1`, and it's a very easy case to catch and handle specifically.

In any case, this test could use a better description. It's not so much testing `find()` with `undefined` options, but rather `find()` without options. Which happens to behave exactly the same as giving it a number.

The assertion/warning about passing in a number (when the user probably wanted to pass a string) is a good start. I'll look for a good place to put the assertion. Using `Ember.assert()` will do, right? That gets removed when building for production.. (wait.. does it? https://github.com/ember-cli/rfcs/pull/50)

Once I get sick of `user.id.toString()` etc, when using the lib we can discuss casting to string again :)